src/rust/bitbox02/src/memory.rs

```diff
@@ -326,4 +406,258 @@ mod tests {
 
         set_salt_root(original.as_slice().try_into().unwrap()).unwrap();
     }
+
+    static RAND_FIXTURES: [[u8; 32]; 7] = [
+        // salt root
+        hex!("bdb9ca4975e59e1b61d9141c5e79688cba7b3989b52b782de2e7e49b07ec8fae"),
+        // io_protection_key
+        hex!("28309e5a2e3bcf4aac94c0e59010fa3492e10839efb5b66192ad18f66a80510b"),
+        // io_protection_key_split
+        hex!("ae5be44d8b71a6041a7e9733e55f8c88b79dd552107624e0a916c10d8755e04e"),
+        // authorization_key
+        hex!("62c741d9ce7832e856ec06f6351cefcd9e7c5ca607938abb709770a5f2dbebcb"),
+        // authorization_key_split
+        hex!("20742d5a582f1f25b6e9d1c1e8b1effb40cfac855667ea7f49968af7f7eb5c19"),
+        // encryption_key
+        hex!("ed183784cbd297f9c2c241d0dd7cd16d62366c44b833ddf2c012fb4b49e1e8f3"),
+        // encryption_key_split
+        hex!("19f60ee825e752150d308817348c0fa6b3fe4f604c85c17e2eb97ada604a476f"),
+    ];
+    static RAND_FIXTURE_INDEX: AtomicUsize = AtomicUsize::new(0);
+
+    unsafe extern "C" fn memory_setup_rand_mock_test_functional(buf_out: *mut u8) {
+        let index = RAND_FIXTURE_INDEX.fetch_add(1, Ordering::SeqCst);
+        let fixture = RAND_FIXTURES.get(index).expect("unexpected RNG request");
+        unsafe { ptr::copy_nonoverlapping(fixture.as_ptr(), buf_out, fixture.len()) };
+    }
+
+    unsafe extern "C" fn mcu_random_adapter(buf_out: *mut u8) {
+        let slice = unsafe { slice::from_raw_parts_mut(buf_out, 32) };
+        let array: &mut [u8; 32] = slice.try_into().unwrap();
+        random::mcu_32_bytes(array);
+    }
+
+    #[test]
+    fn test_memory_multisig() {
+        mock_memory();
+        fake_memory_factoryreset();
+
+        let hashes: [[u8; 32]; 6] = [
+            *b"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+            *b"bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb",
+            *b"cccccccccccccccccccccccccccccccc",
+            *b"dddddddddddddddddddddddddddddddd",
+            *b"eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
+            *b"ffffffffffffffffffffffffffffffff",
+        ];
+        let names = ["name1", "name2", "name3", "name4", "name5", "name6"];
+
+        assert!(multisig_get_by_hash(&hashes[0]).is_none());
+
+        // set
+        assert!(multisig_set_by_hash(&hashes[0], names[0]).is_ok());
+        assert!(multisig_set_by_hash(&hashes[1], names[1]).is_ok());
+        assert!(multisig_set_by_hash(&hashes[1], names[1]).is_ok());
```

Comment:
I think you want to keep the comment here that says "overwrite is possible", otherwise it looks like copy/paste error.

Reply:
Done